helperFunctions/subscribeHelpers/SalesforceAgentAssignedHelpers/SubscribeToLiveAgentHelper.ts

```diff
@@ -69,24 +73,42 @@ export class SubscribeToLiveAgent {
 						if (chatEndedReason === 'agent') {
 							await updateRoomCustomFields(this.data.room.id, { agentEndedChat: true }, this.read, this.modify);
 							this.modify.getAnalytics().sendEvent(getEventData(this.data.room.id, EventName.CHAT_CLOSED_BY_AGENT));
+							await handleEndChatCallback.handleEndChat(CloseMessages.CLOSED_BY_AGENT);
+						} else {
+							await handleEndChatCallback.handleEndChat(CloseMessages.CLOSED_BY_VISITOR);
 						}
-						await handleEndChatCallback.handleEndChat();
+					} else if (isChatRequestFail === true) {
+						const {
+							message: { reason: chatEndedReason },
+						} = getForEvent(messageArray, 'ChatRequestFail');
+						if (chatEndedReason === 'Unavailable') {
+							await updateRoomCustomFields(this.data.room.id, { agentUnavailable: true }, this.read, this.modify);
+							this.modify.getAnalytics().sendEvent(getEventData(this.data.room.id, EventName.CHAT_CLOSED_LIVEAGENT_DISCONNECTED));
+							await handleEndChatCallback.handleEndChat(CloseMessages.LIVEAGENT_DISCONNECTED);
+						}
+						await handleEndChatCallback.handleEndChat(CloseMessages.UNKNOWN_ERROR_IN_CHECKING_AGENT_RESPONSE);
```

Comment:
When `chatEndedReason === 'Unavailable'` we are calling handleChat, and this one is being called every time. We should only call handleEndChat only once when `chatEndedReason === 'Unavailable'`  right?